drivers/volume/portworx/portworx.go

```diff
@@ -844,7 +844,23 @@ func (p *portworx) SnapshotCreate(
 		}
 		snapshotID, err = volDriver.Snapshot(volumeID, true, locator, true)
 		if err != nil {
-			return nil, getErrorSnapshotConditions(err), err
+			// Check already exists error and return existing snapshot if found
+			if isAlreadyExistsError(err) {
```

Comment:
File a bug to return a proper typed error from `volDriver.Snapshot()`.

Reply:
Done PWX-8959